src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/StringAttributeCollection.cs

```diff
@@ -44,7 +44,7 @@ internal void FromString(string list)
                 foreach (string item in items)
                 {
                     string trimmedItem = item.Trim();
-                    if (trimmedItem.Length != 0) Add(item.Trim());
+                    if (trimmedItem.Length != 0) Add(trimmedItem);
```

Comment:
Even looking at the "Add" implementation:

```csharp
        public new void Add(string value)
        {
            ThrowIfReadOnly();
            ThrowIfContainsDelimiter(value);
            _modified = true;
            base.Add(value.Trim());
        }
```

So, the Trim shouldn't be even needed?

Reply:
The second Trim() in `string.Trim().Trim()` is effectively a nop, so there's not a lot of downside in having the call site do the trim.  And if there did happen to be a ton of whitespace that would have been trimmed off, doing it at the call site makes operations in the callee like ThrowIfContainsDelimiter cheaper.